### ISSUE TITLE
Support for #undef

### ISSUE THREAD

**Issue by Liozou** [tecosaur/About.jl#19]
Hey there, thanks for this excellent package!
Do you plan on supporting uninitialized `struct`s ? I'm thinking specifically of the non-`isbits` ones, which can contain `#undef` values such as :
```julia
julia> about(Ref{String}())
Base.RefValue{String} (mutable) (<: Ref{String} <: Any), occupies 8B.
ERROR: UndefRefError: access to undefined reference
Stacktrace:
 [1] memorylayout(io::Base.TTY, value::Base.RefValue{String})
   @ About ~/.julia/packages/About/52iMk/src/values.jl:78
 [2] about(io::Base.TTY, value::Base.RefValue{String})
   @ About ~/.julia/packages/About/52iMk/src/values.jl:43
 [3] about(x::Base.RefValue{String})
   @ About ~/.julia/packages/About/52iMk/src/About.jl:201
 [4] top-level scope
   @ REPL[12]:1
```

Similarly, for `Memory` :
```julia
julia> about(Memory{Any}(undef, 3))
3-element Memory{Any} (mutable) (<: DenseVector{Any} <: AbstractVector{Any} <: Any)
 Memory footprint: 24B directly (referencing 40B in total)
ERROR: Abstract type Any does not have a definite size.
Stacktrace:
 [1] sizeof(x::Type)
   @ Base ./essentials.jl:768
 [2] memorylayout(io::Base.TTY, mem::Memory{Any})
   @ About ~/.julia/packages/About/52iMk/src/values.jl:385
 [3] about(io::Base.TTY, value::Memory{Any})
   @ About ~/.julia/packages/About/52iMk/src/values.jl:43
 [4] about(x::Memory{Any})
   @ About ~/.julia/packages/About/52iMk/src/About.jl:201
 [5] top-level scope
   @ REPL[10]:1
```

**Comment by tecosaur** [comment]
This isn't something I've tested, but it would be good to make it not error if nothing else :)

**Comment by tecosaur** [comment]
This should be fixed as of b4ae600210b348d07eefb758281fe9bd425928d4 :slightly_smiling_face: 

```julia-repl
julia> about(Ref{String}())
Base.RefValue{String} (mutable) (<: Ref{String} <: Any), occupies 8B.
 x::String 8B #undef Uninitialised value

 ■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■
                                   *                                   

 * = Pointer (8B)

julia> about(Memory{Any}(undef, 3))
3-element Memory{Any} (mutable) (<: DenseVector{Any} <: AbstractVector{Any} <: Any), occupies 24B directly (referencing 40B in total)

 Memory block (CPU-addressed) from 0x00007f2b25b97dd0 to 0x00007f2b25b97de8.

 ┌───────────────────────────╴#undef╶───────────────────────────┐ ⋯(×1)⋯ ┌───────────────────────────╴#undef╶───────────────────────────┐  3 items
 0000000000000000000000000000000000000000000000000000000000000000        0000000000000000000000000000000000000000000000000000000000000000    in
 └─0x00─┘└─0x00─┘└─0x00─┘└─0x00─┘└─0x00─┘└─0x00─┘└─0x00─┘└─0x00─┘ ⋯(×8)⋯ └─0x00─┘└─0x00─┘└─0x00─┘└─0x00─┘└─0x00─┘└─0x00─┘└─0x00─┘└─0x00─┘ 24 bytes
```
